mem/mem.go

```diff
@@ -69,17 +89,23 @@ func (mp *memPlugin) GetMetricTypes(_ plugin.ConfigType) ([]plugin.MetricType, e
 // It returns error in case retrieval was not successful
 func (mp *memPlugin) CollectMetrics(metricTypes []plugin.MetricType) ([]plugin.MetricType, error) {
 	metrics := []plugin.MetricType{}
-	getStats(mp.stats)
+
+	pathMemInfo, err := config.GetConfigItem(metricTypes[0], "procfs_path")
+	if err != nil {
+		return nil, err
+	}
+
+	stats := &MemMetrics{}
+	getStats(pathMemInfo.(string), stats)
+
 	for _, metricType := range metricTypes {
-		ns := metricType.Namespace().Strings()
-		if len(ns) < 4 {
-			return nil, fmt.Errorf("Namespace length is too short (len = %d)", len(ns))
-		}
-		stat := ns[3]
-		val, ok := mp.stats[stat]
-		if !ok {
-			return nil, fmt.Errorf("Requested stat %s is not available!", stat)
+		namespace := metricType.Namespace()
+		if len(namespace.Strings()) < 4 {
+			return nil, fmt.Errorf("Namespace length is too short (len = %d)", len(namespace.Strings()))
 		}
+
+		val := ns.GetValueByNamespace(stats, namespace.Strings()[3:])
```

Comment:
I am not seeing where `ns` is declared here. I think the old value you had was called `ns` and now it is called `namespace`. I could be missing something though